### DISCUSSION TITLE
How can I customize the appearance of top tabs (ShellSection) in MAUI for ios

### DISCUSSION THREAD

**Discussion by ParzLaboratory** [dotnet/maui#25091] [Q&A]
### Description

I want to customise the top shell tabs structure  in my maui app for example here in this image we have these two tabs cats and dogs
![two-page-app-top-tabs](https://github.com/user-attachments/assets/1af77859-e1f3-4ad0-873b-75305974e6bd)

I want my shell content top tabs structure to look like this

![Screenshot_20241004_203306_YouTube](https://github.com/user-attachments/assets/a7f637db-3220-411a-8f0f-389f9fc9fff7)

In Android i tried with the help of tabLayoutAppearanceTracker inside the class using CustomShellHandler but for ios I can't find any method like that 
It would be nice if someone suggest any workaround as there's no document regarding this for ios.

**Comment by Kanat9494** [comment]
Hi, I think it is possible with CarouselView and using another elements as CarouselView indicators, everytime changed carouselView index use PositionChanged="OnPositionChanged" and change your element that is responsible for the indicator, I made the same for my login/register page:
<CarouselView x:Name="authTabCV" Loop="False" ItemsSource="{Binding AuthTabs}" ItemTemplate="{StaticResource authUISelector}" 
            PositionChanged="OnPositionChanged" Grid.Row="1" Position="{Binding Index}" HorizontalScrollBarVisibility="Never"/>
            
P.S> you have to use DataTemplateSelectors for each tab

**Comment by davidortinau** [comment]
I would recommend not using the Shell top tabs for this design. Shell is intended for the most common UI scenarios that closely match the native platform UX. You can style it using handlers, but that won't get you the segmented bar look your design has. 

Instead, for that page (or set of pages depending on how you look at it) I recommend looking at these libraries:

- https://github.com/roubachof/Sharpnado.Tabs - probably the most flexible and would meet your needs
- https://help.syncfusion.com/maui-toolkit/tabview/tab-bar-customization - might get you there, but not sure it has the styling needed
- https://www.telerik.com/maui-ui/tabview - also not sure it has the templating and styling needed
- https://docs.devexpress.com/MAUI/404037/dialogs-menu-and-navigation/tab-view/visual-elements

The rest of your app can still use Shell, and if you need deep navigation to these tab children, your tab host page could handle that independently I should think.
